src/mem/packet.hh

```diff
@@ -419,6 +419,8 @@ class Packet : public Printable, public Extensible<Packet>
      */
     uint64_t htmTransactionUid;
 
+    bool _isGPUFuncAccess;
+
```

Comment:
Historically these have gone in the request class.  Does it need to be part of Packet class (i.e., can you have a Packet with a nullptr Request?)

Reply:
To add on here, I would *strongly* prefer not changing Packet *or* Request unless absolutely necessary. There should be a very high bar for changing these widely used objects. Why do we need a special functional access flag for the GPU?

Reply:
I can try to answer based on the comment in RubyPort.cc... This is related to the multiple Networks in one RubySystem which uses the requestor ID to know if requests should go to the GPU or CPU network. It defaults to 0 which is an unmatched network = CPU side. To know the network at the RubyPort, some misc. object accessible by the RubyPort must know which network to pick. Packet/Request have been the best interface so far.

This is mainly because of the way *I* implemented that multiple network support. The multiple-RubySystem supersedes this but would require moving to that...  In the long term everything related to the multiple Networks will need a refactor, I think.  I am aware of some folks using it, but I don't know if they can use multiple-RubySystem. Right now this is the common way to deal with multiple networks in Ruby.

Reply:
The issue I faced when making this fix was that the GPU functional accesses go through network 0 (CPU network) by default. I use this flag to make packets use one of the GPU requestor IDs so that it is injected into network 1 when it reaches RubyPort instead of network 0. I couldn't find any other structure that is shared between Ruby and the GPU compute units that would help me with this.

Reply:
> Historically these have gone in the request class. Does it need to be part of Packet class (i.e., can you have a Packet with a nullptr Request?)

Moved it to the request class.

Reply:
My sense is that @v-ramadas is saying the answer to your question is no, @powerjg .  Or at least absent the complete working @abmerop is proposing, the answer is no.  So unless you all have an alternative, I think this is the best option we could think of.  Let us know ...